src/endpoint/virtualdevices.ts

```diff
@@ -0,0 +1,89 @@
+import { Endpoint} from '../endpoint'
+import { EndpointClient, EndpointClientConfig, HttpClientParams } from '../endpoint-client'
+import { Device, DeviceEvent, VirtualDeviceDetails } from './devices'
+import { DeviceProfileCreateRequest } from './deviceprofiles'
+import { Status, SuccessStatusValue } from '../types'
+
+
+export interface VirtualDeviceOwner {
+	ownerType: VirtualDeviceOwnerTypeEnum
+	ownerId: string
+}
+
+export enum VirtualDeviceOwnerTypeEnum {
+	USER = 'USER',
+	LOCATION = 'LOCATION'
+}
+
+export enum ExecutionTarget {
+	CLOUD = 'CLOUD',
+	LOCAL = 'LOCAL'
+}
+
+export interface VirtualDeviceCreateRequest {
+	name: string
+	owner: VirtualDeviceOwner
+	deviceProfileId?: string
+	deviceProfile?: DeviceProfileCreateRequest
+	roomId?: string
+	executionTarget?: ExecutionTarget
+	hubId?: string
+}
```

Comment:
Fixed, but I didn't extend VirtualDeviceDetails since I don't think there is any guarantee in the face of future changes that the creation request will always contain everything that's shown in the `VirtualDeviceDetails` portion of the device record.